src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -149,6 +181,19 @@ private StatementSyntax Transform( StatementSyntax stmt )
 				.WithCondition( Transform( whileStmt.Condition ) )
 				.WithStatement( Transform( whileStmt.Statement ) ),
 
+			LocalDeclarationStatementSyntax localDeclStmt => localDeclStmt
+				.WithDeclaration( Transform( localDeclStmt.Declaration ) ),
+
+			TryStatementSyntax tryStmt => tryStmt
+				.WithBlock( Transform( tryStmt.Block ) )
+				.WithCatches( TransformAll( tryStmt.Catches, Transform ) ),
+
+			ForEachStatementSyntax forEachStmt => forEachStmt
+				.WithType( TransformType( forEachStmt.Type ) )
+				.WithIdentifier( RemoveAsyncSuffix( forEachStmt.Identifier, optional: true ) )

```

Comment:
Hm, is this like:

```cs
foreach( var fooAsync = ... ) ...
```

into

```cs
foreach( var foo = ... ) ...
```

If so I don't think we need to modify the identifier -- its really unlikely to have an async suffix, but also we're not really un-asyncing it per-se. And like consider that inside the body of the `foreach` unless we modify every identifier to remove the suffixes this would potentially introduce errors, like:

```cs
foreach( var fooAsync = ... ) {
  Bar( fooAsync );
}
```

-->

```cs
foreach( var foo = ... ) {
  Bar( fooAsync ); // error: fooAsync is undefined
}
```

Reply:
I left a comment below in the test case